### ISSUE TITLE
Install on Minikube(docs/install/Knative-with-Minikube.md)

### ISSUE THREAD

**Issue by digikin** [knative/docs#1682]
## Expected Behavior
To install without errors. 

## Actual Behavior
```
kubectl apply --selector knative.dev/crd-install=true \
>    --filename https://github.com/knative/serving/releases/download/v0.7.0/serving.yaml \
>    --filename https://github.com/knative/build/releases/download/v0.7.0/build.yaml \
>    --filename https://github.com/knative/eventing/releases/download/v0.7.0/release.yaml \
>    --filename https://github.com/knative/serving/releases/download/v0.7.0/monitoring.yaml
customresourcedefinition.apiextensions.k8s.io/certificates.networking.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/clusteringresses.networking.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/configurations.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/ingresses.networking.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/podautoscalers.autoscaling.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/revisions.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/routes.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/services.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/serverlessservices.networking.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/builds.build.knative.dev created
customresourcedefinition.apiextensions.k8s.io/buildtemplates.build.knative.dev created
customresourcedefinition.apiextensions.k8s.io/clusterbuildtemplates.build.knative.dev created
customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/apiserversources.sources.eventing.knative.dev created
customresourcedefinition.apiextensions.k8s.io/brokers.eventing.knative.dev created
customresourcedefinition.apiextensions.k8s.io/channels.eventing.knative.dev created
customresourcedefinition.apiextensions.k8s.io/clusterchannelprovisioners.eventing.knative.dev created
customresourcedefinition.apiextensions.k8s.io/containersources.sources.eventing.knative.dev created
customresourcedefinition.apiextensions.k8s.io/cronjobsources.sources.eventing.knative.dev created
customresourcedefinition.apiextensions.k8s.io/eventtypes.eventing.knative.dev created
customresourcedefinition.apiextensions.k8s.io/sequences.messaging.knative.dev created
customresourcedefinition.apiextensions.k8s.io/subscriptions.eventing.knative.dev created
customresourcedefinition.apiextensions.k8s.io/triggers.eventing.knative.dev created
customresourcedefinition.apiextensions.k8s.io/inmemorychannels.messaging.knative.dev created
unable to recognize "https://github.com/knative/serving/releases/download/v0.7.0/serving.yaml": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"
unable to recognize "https://github.com/knative/build/releases/download/v0.7.0/build.yaml": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"
unable to recognize "https://github.com/knative/build/releases/download/v0.7.0/build.yaml": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"
unable to recognize "https://github.com/knative/build/releases/download/v0.7.0/build.yaml": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"
unable to recognize "https://github.com/knative/build/releases/download/v0.7.0/build.yaml": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"
unable to recognize "https://github.com/knative/eventing/releases/download/v0.7.0/release.yaml": no matches for kind "ClusterChannelProvisioner" in version "eventing.knative.dev/v1alpha1"
``` 

## Steps to Reproduce the Problem

1. Install minikube
1. Start kubernetes
1. Install knative per the instructions https://knative.dev/docs/install/knative-with-minikube/#creating-a-kubernetes-cluster

## Additional Info
**Additional context**
The first small recommendation is to put the minikube kubernetes version up to 1.14.0 to be on a more current version.  
```
minikube start --memory=8192 --cpus=6 \
  --kubernetes-version=v1.14.0 \
  --vm-driver=kvm2 \
  --disk-size=30g \
  --extra-config=apiserver.enable-admission-plugins="LimitRanger,NamespaceExists,NamespaceLifecycle,ResourceQuota,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook"  
```
   
I have seen a few issues that have been opened and closed because of installation on minikube.  
I found that the directions do not line up with the commands and available versions in the docs.  
Example: 
1. To install Knative, first install the CRDs `by running the kubectl apply command once with the -l knative.dev/crd-install=true flag`. This prevents race conditions during the install, which cause intermittent errors:
Then the command to issue:
```
 kubectl apply --selector knative.dev/crd-install=true \  
 --filename https://github.com/knative/serving/releases/download/v0.7.0/serving.yaml \  
 --filename https://github.com/knative/build/releases/download/v0.7.0/build.yaml \  
 --filename https://github.com/knative/eventing/releases/download/v0.7.0/release.yaml \  
 --filename https://github.com/knative/serving/releases/download/v0.7.0/monitoring.yaml  
```
### Correction
This fixes the race conditions during the install while eliminating the `no matches for kind "Image" in version`.  
```
 kubectl apply -l knative.dev/crd-install=true \  
--filename https://github.com/knative/serving/releases/download/v0.7.1/serving.yaml \  
--filename https://github.com/knative/build/releases/download/v0.7.0/build.yaml \  
--filename https://github.com/knative/eventing/releases/download/v0.7.0/release.yaml \  
--filename https://github.com/knative/serving/releases/download/v0.7.1/monitoring.yaml  
```
The last command should also match the correct version numbers.  
```
kubectl apply --filename https://github.com/knative/serving/releases/download/v0.7.1  /serving.yaml --selector networking.knative.dev/certificate-provider!=cert-manager \  
--filename https://github.com/knative/build/releases/download/v0.7.0/build.yaml \  
--filename https://github.com/knative/eventing/releases/download/v0.7.0/release.yaml \  
--filename https://github.com/knative/serving/releases/download/v0.7.1/monitoring.yaml  
```
This will install knative without any issue.  

```
kubectl apply -l knative.dev/crd-install=true \
> --filename https://github.com/knative/serving/releases/download/v0.7.1/serving.yaml \
> --filename https://github.com/knative/build/releases/download/v0.7.0/build.yaml \
> --filename https://github.com/knative/eventing/releases/download/v0.7.0/release.yaml \
> --filename https://github.com/knative/serving/releases/download/v0.7.1/monitoring.yaml
customresourcedefinition.apiextensions.k8s.io/certificates.networking.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/clusteringresses.networking.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/ingresses.networking.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/podautoscalers.autoscaling.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/serverlessservices.networking.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/configurations.serving.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/revisions.serving.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/routes.serving.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/services.serving.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/builds.build.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/buildtemplates.build.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/clusterbuildtemplates.build.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/apiserversources.sources.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/brokers.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/channels.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/clusterchannelprovisioners.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/containersources.sources.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/cronjobsources.sources.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/eventtypes.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/sequences.messaging.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/subscriptions.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/triggers.eventing.knative.dev configured
customresourcedefinition.apiextensions.k8s.io/inmemorychannels.messaging.knative.dev configured  


kubectl apply --filename https://github.com/knative/serving/releases/download/v0.7.1/serving.yaml --selector networking.knative.dev/certificate-provider!=cert-manager \
> --filename https://github.com/knative/build/releases/download/v0.7.0/build.yaml \
> --filename https://github.com/knative/eventing/releases/download/v0.7.0/release.yaml \
> --filename https://github.com/knative/serving/releases/download/v0.7.1/monitoring.yaml
namespace/knative-serving created
clusterrole.rbac.authorization.k8s.io/knative-serving-istio created
clusterrole.rbac.authorization.k8s.io/custom-metrics-server-resources created
clusterrole.rbac.authorization.k8s.io/knative-serving-admin created
clusterrole.rbac.authorization.k8s.io/knative-serving-core created
serviceaccount/controller created
clusterrolebinding.rbac.authorization.k8s.io/custom-metrics:system:auth-delegator created
clusterrolebinding.rbac.authorization.k8s.io/hpa-controller-custom-metrics created
clusterrolebinding.rbac.authorization.k8s.io/knative-serving-controller-admin created
rolebinding.rbac.authorization.k8s.io/custom-metrics-auth-reader created
gateway.networking.istio.io/knative-ingress-gateway created
gateway.networking.istio.io/cluster-local-gateway created
customresourcedefinition.apiextensions.k8s.io/certificates.networking.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/clusteringresses.networking.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/ingresses.networking.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/podautoscalers.autoscaling.internal.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/serverlessservices.networking.internal.knative.dev unchanged
service/activator-service created
service/controller created
service/webhook created
image.caching.internal.knative.dev/queue-proxy created
deployment.apps/activator created
service/autoscaler created
deployment.apps/autoscaler created
configmap/config-autoscaler created
configmap/config-defaults created
configmap/config-deployment created
configmap/config-domain created
configmap/config-gc created
configmap/config-istio created
configmap/config-logging created
configmap/config-network created
configmap/config-observability created
configmap/config-tracing created
deployment.apps/controller created
apiservice.apiregistration.k8s.io/v1beta1.custom.metrics.k8s.io created
deployment.apps/networking-istio created
deployment.apps/webhook created
customresourcedefinition.apiextensions.k8s.io/configurations.serving.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/revisions.serving.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/routes.serving.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/services.serving.knative.dev unchanged
namespace/knative-build created
podsecuritypolicy.policy/knative-build created
clusterrole.rbac.authorization.k8s.io/knative-build-admin created
serviceaccount/build-controller created
clusterrolebinding.rbac.authorization.k8s.io/build-controller-admin created
customresourcedefinition.apiextensions.k8s.io/builds.build.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/buildtemplates.build.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/clusterbuildtemplates.build.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev unchanged
service/build-controller created
service/build-webhook created
image.caching.internal.knative.dev/creds-init created
image.caching.internal.knative.dev/git-init created
image.caching.internal.knative.dev/gcs-fetcher created
image.caching.internal.knative.dev/nop created
configmap/config-logging created
configmap/config-observability created
deployment.apps/build-controller created
deployment.apps/build-webhook created
namespace/knative-eventing created
clusterrole.rbac.authorization.k8s.io/addressable-resolver created
clusterrole.rbac.authorization.k8s.io/service-addressable-resolver created
clusterrole.rbac.authorization.k8s.io/serving-addressable-resolver created
clusterrole.rbac.authorization.k8s.io/channel-addressable-resolver created
clusterrole.rbac.authorization.k8s.io/broker-addressable-resolver created
clusterrole.rbac.authorization.k8s.io/eventing-broker-filter created
clusterrole.rbac.authorization.k8s.io/eventing-broker-ingress created
clusterrole.rbac.authorization.k8s.io/eventing-config-reader created
clusterrole.rbac.authorization.k8s.io/channelable-manipulator created
clusterrole.rbac.authorization.k8s.io/knative-eventing-controller created
serviceaccount/eventing-controller created
serviceaccount/eventing-webhook created
clusterrole.rbac.authorization.k8s.io/knative-eventing-webhook created
clusterrolebinding.rbac.authorization.k8s.io/eventing-controller created
clusterrolebinding.rbac.authorization.k8s.io/eventing-controller-resolver created
clusterrolebinding.rbac.authorization.k8s.io/eventing-controller-manipulator created
clusterrolebinding.rbac.authorization.k8s.io/eventing-webhook created
customresourcedefinition.apiextensions.k8s.io/apiserversources.sources.eventing.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/brokers.eventing.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/channels.eventing.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/clusterchannelprovisioners.eventing.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/containersources.sources.eventing.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/cronjobsources.sources.eventing.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/eventtypes.eventing.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/sequences.messaging.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/subscriptions.eventing.knative.dev unchanged
customresourcedefinition.apiextensions.k8s.io/triggers.eventing.knative.dev unchanged
configmap/default-channel-webhook created
service/sources-controller created
service/eventing-webhook created
deployment.apps/eventing-controller created
deployment.apps/sources-controller created
deployment.apps/eventing-webhook created
deployment.apps/webhook created
configmap/config-logging created
configmap/config-observability created
configmap/config-tracing created
clusterrole.rbac.authorization.k8s.io/imc-channelable-manipulator created
clusterrole.rbac.authorization.k8s.io/imc-controller created
clusterrole.rbac.authorization.k8s.io/imc-dispatcher created
service/imc-dispatcher created
serviceaccount/imc-controller created
serviceaccount/imc-dispatcher created
clusterrolebinding.rbac.authorization.k8s.io/imc-controller created
clusterrolebinding.rbac.authorization.k8s.io/imc-dispatcher created
customresourcedefinition.apiextensions.k8s.io/inmemorychannels.messaging.knative.dev unchanged
deployment.apps/imc-controller created
deployment.apps/imc-dispatcher created
clusterchannelprovisioner.eventing.knative.dev/in-memory created
serviceaccount/in-memory-channel-controller created
clusterrole.rbac.authorization.k8s.io/in-memory-channel-controller created
clusterrolebinding.rbac.authorization.k8s.io/in-memory-channel-controller created
deployment.apps/in-memory-channel-controller created
serviceaccount/in-memory-channel-dispatcher created
clusterrole.rbac.authorization.k8s.io/in-memory-channel-dispatcher created
clusterrolebinding.rbac.authorization.k8s.io/in-memory-channel-dispatcher created
role.rbac.authorization.k8s.io/in-memory-channel-dispatcher created
rolebinding.rbac.authorization.k8s.io/in-memory-channel-dispatcher created
deployment.apps/in-memory-channel-dispatcher created
namespace/knative-monitoring created
service/elasticsearch-logging created
serviceaccount/elasticsearch-logging created
clusterrole.rbac.authorization.k8s.io/elasticsearch-logging created
clusterrolebinding.rbac.authorization.k8s.io/elasticsearch-logging created
statefulset.apps/elasticsearch-logging created
service/kibana-logging created
deployment.apps/kibana-logging created
configmap/fluentd-ds-config created
serviceaccount/fluentd-ds created
clusterrole.rbac.authorization.k8s.io/fluentd-ds created
clusterrolebinding.rbac.authorization.k8s.io/fluentd-ds created
service/fluentd-ds created
daemonset.apps/fluentd-ds created
serviceaccount/kube-state-metrics created
role.rbac.authorization.k8s.io/kube-state-metrics-resizer created
rolebinding.rbac.authorization.k8s.io/kube-state-metrics created
clusterrole.rbac.authorization.k8s.io/kube-state-metrics created
clusterrolebinding.rbac.authorization.k8s.io/kube-state-metrics created
deployment.extensions/kube-state-metrics created
service/kube-state-metrics created
configmap/grafana-dashboard-definition-kubernetes-deployment created
configmap/grafana-dashboard-definition-kubernetes-capacity-planning created
configmap/grafana-dashboard-definition-kubernetes-cluster-health created
configmap/grafana-dashboard-definition-kubernetes-cluster-status created
configmap/grafana-dashboard-definition-kubernetes-control-plane-status created
configmap/grafana-dashboard-definition-kubernetes-resource-requests created
configmap/grafana-dashboard-definition-kubernetes-nodes created
configmap/grafana-dashboard-definition-kubernetes-pods created
configmap/grafana-dashboard-definition-kubernetes-statefulset created
serviceaccount/node-exporter created
clusterrole.rbac.authorization.k8s.io/node-exporter created
clusterrolebinding.rbac.authorization.k8s.io/node-exporter created
daemonset.extensions/node-exporter created
service/node-exporter created
configmap/grafana-dashboard-definition-knative-efficiency created
configmap/grafana-dashboard-definition-knative-reconciler created
configmap/scaling-config created
configmap/grafana-dashboard-definition-knative created
configmap/grafana-datasources created
configmap/grafana-dashboards created
service/grafana created
deployment.apps/grafana created
configmap/prometheus-scrape-config created
service/kube-controller-manager created
service/prometheus-system-discovery created
serviceaccount/prometheus-system created
role.rbac.authorization.k8s.io/prometheus-system created
role.rbac.authorization.k8s.io/prometheus-system created
role.rbac.authorization.k8s.io/prometheus-system created
role.rbac.authorization.k8s.io/prometheus-system created
clusterrole.rbac.authorization.k8s.io/prometheus-system created
rolebinding.rbac.authorization.k8s.io/prometheus-system created
rolebinding.rbac.authorization.k8s.io/prometheus-system created
rolebinding.rbac.authorization.k8s.io/prometheus-system created
rolebinding.rbac.authorization.k8s.io/prometheus-system created
clusterrolebinding.rbac.authorization.k8s.io/prometheus-system created
service/prometheus-system-np created
statefulset.apps/prometheus-system created
service/zipkin created
deployment.apps/zipkin created
```

**Install information:**
 - Platform (GKE, IKS, AKS, etc.): minikube v 1.3
 - Knative Version: 0.7

I can create a pull request if its ok. Or will this be updated in v0.8?

**Comment by digikin** [comment]
Sorry for changing the status I guess this did go stale now that version 0.13 is out but I can go over the documentation again to double check.

**Comment by vugardzhamalov** [comment]
Hi @digikin! I am not sure if I should open a new ticket, following error was coming up for me during v.014.0 installation on Minikube: 
`error: unable to recognize "https://github.com/knative/serving/releases/download/v0.14.0/serving-core.yaml": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"`

In order to resolve (both for Serving and Eventing components) I had to execute twice!!! (for each of the following):
`kubectl apply --selector knative.dev/crd-install=true \
         --filename https://github.com/knative/serving/releases/download/v0.14.0/serving-crds.yaml \
         --filename https://github.com/knative/serving/releases/download/v0.14.0/serving-core.yaml`

`kubectl apply --selector knative.dev/crd-install=true \
        --filename https://github.com/knative/eventing/releases/download/v0.14.0/eventing-crds.yaml \
        --filename https://github.com/knative/eventing/releases/download/v0.14.0/eventing-core.yaml`

I take it as this issue was expected to be resolved as per discussion here: #https://github.com/knative/docs/issues/1036

**Comment by digikin** [comment]
Honestly I don't have an answer @vagababov nobody at knative seems to answer or respond to this issue. So, I would start a new issue you might have better luck getting someone to at least reply.

**Comment by vagababov** [comment]
@vugardzhamalov 
I think :)

**Comment by jonatasbaldin** [comment]
Hi @vugardzhamalov @digikin @vagababov, how are y'all doing?

Just tested installing v0.14.0 on Minikube and everything worked ✨. I used the instructions from [my blog post](https://deployeveryday.com/2020/03/31/running-knative-locally.html), using Kourier:

```bash
minikube start --cpus=4 --memory=4096 --addons=ingress

# Install the Knative Serving
export KNATIVE_VERSION="0.14.0"
kubectl apply --filename "https://github.com/knative/serving/releases/download/v$KNATIVE_VERSION/serving-crds.yaml"
kubectl apply --filename "https://github.com/knative/serving/releases/download/v$KNATIVE_VERSION/serving-core.yaml"

# Configure the magic xip.io DNS name
kubectl apply --filename "https://github.com/knative/serving/releases/download/v$KNATIVE_VERSION/serving-default-domain.yaml"

# Install and configure Kourier
kubectl apply --filename https://raw.githubusercontent.com/knative/serving/v$KNATIVE_VERSION/third_party/kourier-latest/kourier.yaml
kubectl patch configmap/config-network --namespace knative-serving --type merge --patch '{"data":{"ingress.class":"kourier.ingress.networking.knative.dev"}}'

# Deploy sample service
kubectl apply -f https://gist.githubusercontent.com/jonatasbaldin/bc04de2e376be23f75bb5815041fdd61/raw/d2345ac9aa01d0f3c771e9b3d4a1421dd766e0f9/service.yaml
```

Before accessing the Service, go in another terminal and open the Minikube tunnel:
```bash
minikube tunnel
```

Tested by curling the the Service on its external URL (something like `http://helloworld-go.default.10.96.154.27.xip.io`) and it worked.

This is my Minikube and OS version:
```
minikube version: v1.6.2
commit: 54f28ac5d3a815d1196cd5d57d707439ee4bb392

MacOS Catalina 10.15.3
```

Could you test these instructions? If it doesn't work, could you give us more details about your minikube version, operational system etc?

**Comment by vugardzhamalov** [comment]
Thank you guys @digikin @vagababov @jonatasbaldin !

My minikube profile is rather different but I did it as you've instructed in the earlier post. Not sure if this is of any help to anyone - at this stage I can only point out three things:

1. The issue with 'images.caching.internal.knative.dev' CRD and/or with 'queue-proxy' Image object (and of course the error message I've posted earlier) is definitely still present and it could take up to 30 min to go away but this is intermittent and sometimes it won't get resolved at all - resulting in my inability to proceed with deployment - I've seen a similar issue reported as being intermittent earlier, e.g. here #https://github.com/knative/serving/issues/4176

2. On installation documentation - it seems as if the same set of CRDs has been repeated in 'serving-core.yaml'... I've only glanced through but I see no difference with CRDs in 'serving-crds.yaml'... though to be honest I had no chance to run a diff on these two. Anyone knows if both set of CRDs will be required for some reason? Or should it be enough to execute something like this in order to deploy CRDs and repeat afterwards without '--selector' switch to deploy the rest of the objects?

`kubectl apply --selector knative.dev/crd-install=true --filename https://github.com/knative/serving/releases/download/v0.14.0/serving-core.yaml`

3. It is not very clear if following '--extra-config=' minikube switch value mentioned in v0.12 installation docs is still required for v0.14 and above, anyone knows the definitive answer? Looking at minikube example provided by @jonatasbaldin  in the earlier post - this is probably not required any more... or is it? Thank you!

`--extra-config=apiserver.enable-admission-plugins="LimitRanger,NamespaceExists,NamespaceLifecycle,ResourceQuota,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook"`

And in case if might help, I am running things on:
OS:  Fedora 31
Minikube: minikube version: v1.9.2
cri-o:                1.18 [e]                default [i]

Minikube profile I normally use for Knative deployments:
`minikube start \
    --profile <profile-name> \
    --kubernetes-version=v1.18.0 \
    --vm-driver kvm2 \
    --network-plugin=cni \
    --enable-default-cni \
    --container-runtime=crio \
    --cri-socket='/var/run/crio/crio.sock' \
    --extra-config=kubelet.container-runtime=remote \
    --extra-config=kubelet.container-runtime-endpoint=/var/run/crio.sock \
    --extra-config=kubelet.image-service-endpoint=/var/run/crio.sock \
    --bootstrapper=kubeadm \
    --memory='16384mb' --cpus='6' \
    --disk-size 30g \
    --addons='metrics-server'`

**Comment by vugardzhamalov** [comment]
Oh... I wonder if this #https://github.com/knative/serving/issues/7755 is of any relevance? Seems like it is a similar issue the difference is that it is a nightly instead of v0.14 in my case...

**Comment by jonatasbaldin** [comment]
I'll try to take a look into this this week, but I wasn't able to quickly reproduce it :(

**Comment by vugardzhamalov** [comment]
Hi  @jonatasbaldin 
FYI please consider this as a follow up to my first point in https://github.com/knative/docs/issues/1682#issuecomment-625218932

I've decided to proceed with a clean install of Fedora 32 (the issue was opened while I was still on Fedora 31). And it seems like I don't get any delays with 'Image' object anymore... Not sure what could've been the problem (may somehow could've been related to my host being set to a time zone different from my actual time zone local? - anyways this is fixed on my new deployment), I've realized that I was having similar issues with CRDs during istio-operator and istio configuration profile(s) deployment - so that made me suspicious. So far things looking much better.

Host OS: Fedora 32
Libvirt: 6.1.0
Minikube: v1.10.1
kubectl version --client=true
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:56:40Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

**Comment by abrennan89** [comment]
@digikin the latest docs for 0.14 no longer mention installing on specific platforms, e.g. Minikube, just a generic install guide.
I think this issue is maybe one for the main repo rather than docs if it's about installation issues on Minikube, rather than a specific docs change.
Thanks

**Comment by mattmoor** [comment]
tl;dr if `--selector knative.dev/crd-install=true` is used with anything but `ko`, it is a bug, our public docs should stop referencing it.

`kubectl apply --selector knative.dev/crd-install=true` shouldn't be use anywhere anymore, we should be replacing it with explicit installation of the CRD yamls (e.g. `serving-crds.yaml`).

**AI: It looks like [Serving](https://knative.dev/docs/install/any-kubernetes-cluster/#installing-the-serving-component) fixed this, but [eventing](https://knative.dev/docs/install/any-kubernetes-cluster/#installing-the-eventing-component) didn't** 


_(longer explanation of the actual issue)_

IIRC `kubectl` is not properly respecting `--selector` for instances of CRDs, so the trick we added  for using `crd-install=true` never totally worked (it probably made things worse) because any custom resources installed would blow up like this instead of being filtered.  We see the same thing with the Istio Gateway resources in serving, if you try to purely use `--selector` to avoid Istio.

We actually fixed this in `ko` so that we could avoid both build and deploy, and so we are able to use the selectors to produce the filtered yamls with `ko resolve --selector ... -f config > foo-crds.yaml`.
